src/main/java/frc/robot/Plunger.java

```diff
@@ -2,29 +2,224 @@
 
 //imports
 import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj.AnalogInput;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.Timer;
 
 /**
  *  Controls plunger mechanism.
  */
 public class Plunger{
 
     // object declaration
-    Solenoid solenoid;
+    Solenoid upstreamSolenoid;
+    Solenoid downstreamSolenoid;
+    AnalogInput pressureSensor;
+    AnalogInput vacuumSensor;
+    plungerState state;
+    Timer timer;
+    Joystick joy;
     
+
+    //Solenoid channels
+    public static final int UPSTREAM_SOLENOID_CHANNEL = 0;
+    public static final int DOWNSTREAM_SOLENOID_CHANNEL = 0;
+    public static final int PRESSURE_SENSOR_CHANNEL = 0;
+    public static final int VACUUM_SENSOR_CHANNEL = 0;
+
+    //Solenoid sensor variables
+    public static final double VACUUM_SENSOR_GOOD_VAC = 20;
+    public static final double VACUUM_SENSOR_MIN_VAC = 15; 
+
+    //joystick controls
+    public static final int PLUNGER_SUCTION_BUTTON = 0;
+
     /**
      * Constructs a new plunger object.
      */
-    public Plunger(){
+    public Plunger(DriveBase drive){
 
         // object initialization
-        solenoid = new Solenoid(0);
+        joy = drive.joyR;
```

Comment:
DON'T DO THIS!!!! You should never be accessing other class objects/variables without an accessor method. I'll explain to you what this is today, but don't ever use this object.classVariable format. It's not good practice since you might need to modify the value/characteristics/output of the object or variable before you send data to other classes. Also, you shouldn't be using Joysticks outside of Robot.java.